plugins/module_utils/network/sonic/config/dhcp_relay/dhcp_relay.py

```diff
@@ -0,0 +1,490 @@
+            url = self.dhcpv6_relay_intf_config_path['vrf_select'].format(intf_name=name)
+            requests.append({'path': url, 'method': PATCH, 'data': payload})
+
+        return requests
+
+    def get_delete_dhcp_dhcpv6_relay_completely_requests(self, have):
+        """Get requests to delete all existing DHCP and DHCPv6 relay
+        configurations in the chassis
+        """
+        requests = []
+        for cfg in have:
+            if cfg.get('ipv4'):
+                requests.append({'path': self.dhcp_relay_intf_config_path['server_addresses_all'].format(intf_name=cfg['name']), 'method': DELETE})
+            if cfg.get('ipv6'):
+                requests.append({'path': self.dhcpv6_relay_intf_config_path['server_addresses_all'].format(intf_name=cfg['name']), 'method': DELETE})
+
+        return requests
+
+    def get_delete_dhcp_dhcpv6_relay_requests(self, commands, have):
+        """Get requests to delete DHCP and DHCPv6 relay configurations
+        based on the commands specified
+        """
+        requests = []
+
+        for command in commands:
+            intf_name = command['name']
+            have_obj = next((cfg for cfg in have if cfg['name'] == intf_name), None)
+            if not have_obj:
+                continue
+
+            have_ipv4 = have_obj.get('ipv4')
+            have_ipv6 = have_obj.get('ipv6')
+
+            ipv4 = command.get('ipv4')
+            ipv6 = command.get('ipv6')
+            if intf_name and not ipv4 and not ipv6:
+                if have_ipv4:
+                    requests.append({'path': self.dhcp_relay_intf_config_path['server_addresses_all'].format(intf_name=intf_name), 'method': DELETE})
+                if have_ipv6:
+                    requests.append({'path': self.dhcpv6_relay_intf_config_path['server_addresses_all'].format(intf_name=intf_name), 'method': DELETE})
+            else:
+                if ipv4 and have_ipv4:
+                    requests.extend(self.get_delete_specific_dhcp_relay_param_requests(command, have_obj))
+                if ipv6 and have_ipv6:
+                    requests.extend(self.get_delete_specific_dhcpv6_relay_param_requests(command, have_obj))
+
+        return requests
+
+    def get_delete_specific_dhcp_relay_param_requests(self, command, config):
+        """Get requests to delete specific DHCP relay configurations
+        based on the command specified for the interface
+        """
+        requests = []
+
+        name = command['name']
+        ipv4 = command.get('ipv4')
+        have_ipv4 = config.get('ipv4')
+        if not ipv4 or not have_ipv4:
+            return requests
+
+        server_addresses = self.get_server_addresses(ipv4.get('server_addresses'))
+        have_server_addresses = self.get_server_addresses(have_ipv4.get('server_addresses'))
+
+        # Delete all DHCP relay config for an interface, if only
+        # an address with no value is specified.
```

Comment:
Please add an additional explanatory sentence here and for the corresponding DHCPv6 block:

e.g.

This "special" YAML sequence is supported to provide "delete all AF parameters" functionality despite the Ansible infrastructure limitations that prevent use of a simpler syntax for deleting an entire AF parameter dictionary.

Reply:
Comment has been updated

Reply:
Thanks.

Resolved.